### ISSUE TITLE
possible false positive with RET504 and multiple-assignements

### ISSUE THREAD

**Issue by Borda** [astral-sh/ruff#4236]
I have the following code sample, and here I do think it is correct:
```py
    def consume_result(self) -> T:
        if self._result is None:
            raise MisconfigurationException()
        result, self._result = self._result, None  # free memory
        return result
```

**Comment by charliermarsh** [comment]
Yeah makes sense. This rule is so unreliable, it's just really hard to avoid false positives while retaining much utility.

**Comment by MichaReiser** [comment]
> Yeah makes sense. This rule is so unreliable, it's just really hard to avoid false positives while retaining much utility.

I agree that this code per se is correct and we should, if possible, detect and support it. But this can also be a case where a slight restructuring of the code fixes the false-positive (by splitting the code into two assignments)

**Comment by Borda** [comment]
> But this can also be a case where a slight restructuring of the code fixes the false-positive (by splitting the code into two assignments)

Yes, but then you drop the beauty of puthon and memory efficiency too... :)
(in the end all can be just C code :chipmunk:)
